### PR TITLE
Refactor everest tmpdir fixture

### LINKED ISSUE #8860: Update everest tmpdir functionality
Remove `def tmpdir(path, teardown=True)` and `def tmp(path=None, teardown=True):` from `tests/everest/utils/__init__.py` and replace it with `tmpdir` fixture from pytest.

### PR DESCRIPTION
**Pull request by frode-aarstad**
**Issue**
Resolves #8860 

**Approach**
_Short description of the approach_

(Screenshot of new behavior in GUI if applicable)

- [ ] PR title captures the intent of the changes, and is fitting for release notes.
- [ ] Added appropriate release note label
- [ ] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [ ] Make sure unit tests pass locally after every commit (`git rebase -i main
      --exec 'pytest tests/ert/unit_tests -n logical -m "not integration_test"'`)

## When applicable
- [ ] **When there are user facing changes**: Updated documentation
- [ ] **New behavior or changes to existing untested code**: Ensured that unit tests are added (See [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)).
- [ ] **Large PR**: Prepare changes in small commits for more convenient review
- [ ] **Bug fix**: Add regression test for the bug
- [ ] **Bug fix**: Create Backport PR to latest release

### CODE REVIEW

File: tests/everest/conftest.py [hunk 1]
```diff
@@ -82,3 +83,43 @@ def control_config(
...
+
+@pytest.fixture
+def copy_test_data_to_tmp(tmp_path, monkeypatch):
+    path = relpath("test_data")
+    shutil.copytree(path, tmp_path, dirs_exist_ok=True)
+    monkeypatch.chdir(tmp_path)
+
+
+@pytest.fixture
+def copy_templete_test_data_to_tmp(tmp_path, monkeypatch):
```
Comment (verveerpj) [inline tests/everest/conftest.py:110]: `templete` -> `template`

File: tests/everest/test_everserver.py [hunk 2]
```diff
@@ -121,8 +117,9 @@ def test_everserver_status_failure(mf_1):
     "everest.detached.jobs.everserver.export_to_csv",
     side_effect=partial(check_status, status=ServerStatus.exporting_to_csv),
 )
-@tmpdir(relpath("..", "..", "test-data", "everest", "math_func"))
-def test_everserver_status_running_complete(*args):
+def test_everserver_status_running_complete(
+    _1, _2, _3, _4, _5, _6, _7, _8, _9, copy_math_func_test_data_to_tmp
```
Comment (verveerpj) [inline tests/everest/test_everserver.py:121]: would this work?: `copy_math_func_test_data_to_tmp, *args`. Not sure if that is valid Python... Would it be desirable?

**Review by verveerpj — APPROVED** [review]
LGTM. Small comments.

File: tests/everest/test_everserver.py (hunk 2 above)
Comment (frode-aarstad) [reply tests/everest/test_everserver.py:121]: I was not able to get any version with mixing fixtures and *args as parameters that's why I resorted to _*. I agree its ugly but I found no other way

**Pull request merged**